up42/workflow.py

```diff
@@ -468,6 +471,81 @@ def _helper_run_job(
             job.track_status()
         return job
 
+    def _helper_run_parallel_jobs(
+        self,
+        input_parameters_list: List[Dict] = None,
+        max_concurrent_jobs: int = 10,
+        test_job=False,
+        name: str = None,
+    ) -> "JobCollection":
+        """
+        Helper function to create and run parallel real or test jobs.
+
+        Args:
+            input_parameters_list: List of dictionary of input parameters.
+            max_concurrent_jobs: Maximum number of parallel jobs that can be triggered.
+            test_job: If set, runs a test query (search for available imagery based on your data parameters).
+            name: The job name. Optional, by default the workflow name is assigned.
+
+        Returns:
+            The spawned real or test job object.
+        """
+        if input_parameters_list is None:
+            raise ValueError(
+                "Provide the job parameters via `input_parameters_list`."
+                " You can use workflow.construct_parallel_parameters()!"
+            )
+
+        if test_job:
+            input_parameters_list = copy.deepcopy(input_parameters_list)
+            for input_parameters in input_parameters_list:
+                input_parameters.update({"config": {"mode": "DRY_RUN"}})  # type: ignore
+                logger.info("+++++++++++++++++++++++++++++++++")
+                logger.info("Running this job as Test Query...")
+                logger.info("+++++++++++++++++++++++++++++++++")
+
+        def grouper(iterable, n, fillvalue=None):
+            args = [iter(iterable)] * n
+            return zip_longest(*args, fillvalue=fillvalue)
+
+        jobs_list = []
+        job_nr = 0
+        # Run all jobs in parallel batches of the max_concurrent_jobs (max. 10.)
+        batches = grouper(input_parameters_list, max_concurrent_jobs)
+        for batch in batches:
+            batch_list = [x for x in batch if x]
```

Comment:
Seeing that grouper doesnt work without the fillvalue, the basic range solution (answer 2) is probably the most basic and clearest. Tested it.

        batches = [input_parameters_list[pos:pos + max_concurrent_jobs] for pos in range(0, len(input_parameters_list), max_concurrent_jobs)]

        for batch in batches:
            batch_jobs = []